src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  let copyState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+      resultArray.push(Object.assign({}, copyState));
```

Comment:
[DONT REPEAT YOURSELF]: Instead of pushing the state copy to `resultArray` in every case, do it just once at the end of the loop after processing the current action.